src/DynamoCoreWpf/ViewModels/PackageManager/PackagePathViewModel.cs

```diff
@@ -118,15 +96,21 @@ public PackagePathViewModel(IPreferences setting)
             InitializeRootLocations();
 
             AddPathCommand = new DelegateCommand(p => InsertPath());
-            DeletePathCommand = new DelegateCommand(p => RemovePathAt((int)p), CanDelete);
-            MovePathUpCommand = new DelegateCommand(p => SwapPath((int)p, ((int)p) - 1), CanMoveUp);
-            MovePathDownCommand = new DelegateCommand(p => SwapPath((int)p, ((int)p) + 1), CanMoveDown);
-            UpdatePathCommand = new DelegateCommand(p => UpdatePathAt((int)p), CanUpdate);
+            DeletePathCommand = new DelegateCommand(p => RemovePathAt(ConvertPathToIndex(p)), p=>CanDelete(ConvertPathToIndex(p)));
+            MovePathUpCommand = new DelegateCommand(p => SwapPath(ConvertPathToIndex(p), ConvertPathToIndex(p)-1), p=> CanMoveUp(ConvertPathToIndex(p)));
+            MovePathDownCommand = new DelegateCommand(p => SwapPath(ConvertPathToIndex(p), ConvertPathToIndex(p) + 1), p => CanMoveDown(ConvertPathToIndex(p)));
+            UpdatePathCommand = new DelegateCommand(p => UpdatePathAt(ConvertPathToIndex(p)), p=>CanUpdate(ConvertPathToIndex(p)));
```

Comment:
I think it would be better to refactor these lines of code into a single method and call it from both constructors if they are the same lines of code.

Reply:
sounds good, will do.